jmaqs-webservices-jdk11/src/main/java/com/magenic/jmaqs/webservices/jdk11/WebServiceDriver.java

```diff
@@ -17,38 +22,141 @@
    */
   private HttpClient baseHttpClient;
 
+  /**
+   * the base HTTP request control.
+   */
+  private HttpRequest baseHttpRequest;
+
   /**
    * The URI to be stored for the Web Service.
    */
   private URI baseAddress;
 
   /**
-   * Constructor.
-   *
-   * @param baseAddress
-   *          The base URI address to use
-   * @throws URISyntaxException
-   *           URI syntax is invalid
+   * Class Constructor that sets the http Client.
+   * @param newHttpClient the http client to be set.
    */
-  public WebServiceDriver(String baseAddress) throws URISyntaxException {
-    this(new URI(baseAddress));
+  public WebServiceDriver(HttpClient newHttpClient) {
+    this.baseHttpClient = newHttpClient;
   }
 
   /**
-   * Constructor.
-   *
-   * @param baseAddress
-   *          The base URI address to use
+   * Class constructor that sets the HttpRequest.
+   * @param newHttpRequest the new Http request to be set
    */
-  public WebServiceDriver(URI baseAddress) {
-    this.baseAddress = baseAddress;
+  public WebServiceDriver(HttpRequest newHttpRequest) {
+    this.baseHttpClient = HttpClientFactory.getDefaultClient();
+    this.baseHttpRequest = newHttpRequest;
   }
 
   /**
-   * Class Constructor that sets the http Client.
-   * @param newHttpClient the http client to be set.
+   * Class Constructor that sets the base address as a URI.
+   * @param newBaseAddress The base URI address to use
    */
-  public WebServiceDriver(HttpClient newHttpClient) {
-    this.baseHttpClient = newHttpClient;
+  public WebServiceDriver(URI newBaseAddress) {
+    this.baseHttpClient = HttpClientFactory.getDefaultClient();
+    this.baseHttpRequest = HttpRequestFactory.getRequest(newBaseAddress.toString());
+  }
+
+  /**
+   * Class Constructor that sets the base address as a string.
+   * @param newBaseAddress The base URI address to use
+   * @throws URISyntaxException URI syntax is invalid
+   */
+  public WebServiceDriver(String newBaseAddress) throws URISyntaxException {
+    this(new URI(newBaseAddress));
+  }
+
+  /**
+   * Sets http client.
+   * @param httpClient the http client
+   */
+  public void setHttpClient(HttpClient httpClient) {
+    this.baseHttpClient = httpClient;
+  }
+
+  /**
+   * Gets http client.
+   * @param mediaType the media type
+   * @return the http client
+   */
+  public HttpClient getHttpClient(MediaType mediaType) {
```

Comment:
Why does this require a media type if it's only returning the basehttpclient here?

Reply:
deleted the MediaType parameter